mailpoet/lib/Automation/Integrations/WordPress/Fields/PostFieldsFactory.php

```diff
@@ -0,0 +1,241 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Automation\Integrations\WordPress\Fields;
+
+use MailPoet\Automation\Engine\Data\Field;
+use MailPoet\Automation\Engine\WordPress;
+use MailPoet\Automation\Integrations\WordPress\Payloads\PostPayload;
+
+class PostFieldsFactory {
+
+  /** @var WordPress */
+  private $wp;
+
+  public function __construct(
+    WordPress $wp
+  ) {
+    $this->wp = $wp;
+  }
+
+  /**
+   * @return Field[]
+   */
+  public function getFields(): array {
+    return [
+      new Field(
+        'wordpress:post:id',
+        Field::TYPE_INTEGER,
+        __('ID', 'mailpoet'),
+        function (PostPayload $payload) {
+          return $payload->getPostId();
+        }
+      ),
+      new Field(
+        'wordpress:post:type',
+        Field::TYPE_ENUM,
+        __('Post type', 'mailpoet'),
+        function (PostPayload $payload) {
+          $post = $payload->getPost();
+          //phpcs:ignore Squiz.NamingConventions.ValidVariableName.MemberNotCamelCaps
+          return $post ? $post->post_type : '';
```

Comment:
In many cases, the empty value here is `''`. Is that just for PHPStan, or can it really be empty (in which case I think `null` could be more correct)?